pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -151,22 +151,90 @@ type DNSProxy struct {
 	rejectReply int32
 }
 
+type matchNameMatcher struct {
+	mapping map[string]struct{}
+	key     *string
+}
```

Comment:
Can you leave more comments in this structure? It's not entirely clear to me what are those two fields.

Reply:
Thanks for looking!

Yee, will do! The actual value isn't too important, as long as every "reference" is released, since it has the key.


Right now its just all the domains joined with a `|` (bar)
https://github.com/cilium/cilium/pull/20396/files#diff-a7f128716bd349970eb11e4decfff9d79cb9daa1067020b9c9a7efd664bd8034R383

We might want to improve it later to reduce the number of allocs, but its not too important now imo.

Reply:
Will try to post an updated version this weekend or early next week